### ISSUE TITLE
Intermittent test failure in `timer` test

### ISSUE THREAD

**Issue by Keno** [mozilla/rr#1965]
Just saw this locally:
```
579: Test 'timer' FAILED: : error during recording:
579: --------------------------------------------------
579: [FATAL /data/keno/rr/src/RecordSession.cc:1148:preinject_signal() errno: SUCCESS]
579:  (task 12151 (rec:12151) at time 78183)
579:  -> Assertion `old_ip == t->ip()' failed to hold.
579: [FATAL /data/keno/rr/src/log.cc:284:emergency_debug() errno: SUCCESS] (session doesn't look interactive, aborting emergency debugging)
579: --------------------------------------------------
579: record.out:
579: --------------------------------------------------
579: --------------------------------------------------
579: Test timer failed, leaving behind /tmp/rr-test-timer-osvCQp6l0
579: To replay the failed test, run
579:   _RR_TRACE_DIR=/tmp/rr-test-timer-osvCQp6l0 rr replay
```

**Comment by Keno** [comment]
Haven't been able to reproduce this unfortunately, so the above is all I have. @rocallahan looks like the tried to inject the signal, but didn't end up in a signal stop. Any speculations?

**Comment by rocallahan** [comment]
Knowing the old and new IPs would be helpful. I'll push a patch to master to log that. Other than that I don't really know what could be going on... Is it possible that the tgkill could race with the ptrace_cont in such a way that the tracee's CPU doesn't know there's a signal pending when it resumes? I kinda doubt it.

**Comment by rocallahan** [comment]
https://github.com/mozilla/rr/commit/9245fb5cf93355ef6ebb2dc673c2230c660d4678 will dump more information if this happens again.

**Comment by rocallahan** [comment]
> I kinda doubt it.

Especially because rr and the tracee should be bound to the same CPU...

**Comment by Keno** [comment]
I thought about that as well, but it seems unlikely. The tgkill syscall does not return until the signal is queued.

**Comment by Keno** [comment]
Caught it again (yay for big machines):
```
[FATAL /data/keno/rr/src/RecordSession.cc:1148:preinject_signal() errno: SUCCESS]
 (task 63895 (rec:63895) at time 200850)
 -> Assertion `old_ip == t->ip()' failed to hold. Singlestep actually advanced when we just expected a signal; was at 0x55ec467561c1 now at 0x55ec467561c2 with status 0x57f (STOP-SIGTRAP)
[FATAL /data/keno/rr/src/log.cc:284:emergency_debug() errno: SUCCESS] (session doesn't look interactive, aborting emergency debugging)
```

**Comment by Keno** [comment]
The tail of the log:
```
{
  real_time:2427115.112612 global_time:200840, event:`SYSCALL: rt_sigreturn' (state:EXITING_SYSCALL) tid:63895, ticks:86254270
 rax:0xf997 rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x0 rdi:0x0 rbp:0xf997 rsp:0x681ffe20 r8:0x7ffc954c1ba0 r9:0x7ffc954c1ba0 r10:0x7ffc954c1ba0 r11:0x246 r12:0x55ec46755d60 r13:0x7ffc954c1c80 r14:0x1 r15:0x1 rip:0x70000005 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0xffffffffffffffff fs_base:0x7f3b01302440 gs_base:0x0 st0:0x0 st1:0x0 st2:0x0 st3:0x0 st4:0x0 st5:0x0 st6:0x0 st7:0x0ymm0:0x0 ymm1:0x25252525252525252525252525252525 ymm2:0x687470004655424c4c41435359535f45 ymm3:0x0 ymm4:0xffffffffffffff00 ymm5:0x0 ymm6:0xffffffffffffffffffffffffffffffff ymm7:0x4000000040000000400000004 ymm8:0x7f3b00aa02d800007f3b00aa02d8 ymm9:0x0 ymm10:0x0 ymm11:0x0 ymm12:0x0 ymm13:0x0 ymm14:0x0 ymm15:0x0
}
{
  real_time:2427115.112618 global_time:200841, event:`EXIT_SIGHANDLER' tid:63895, ticks:86254270
}
{
  real_time:2427115.117025 global_time:200842, event:`SYSCALLBUF_FLUSH' tid:63895, ticks:86307082
}
{
  real_time:2427115.117034 global_time:200843, event:`SYSCALL: rrcall_notify_syscall_hook_exit' (state:ENTERING_SYSCALL) tid:63895, ticks:86307082
 rax:0xffffffffffffffda rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x0 rdi:0x0 rbp:0xf997 rsp:0x681ffe20 r8:0x7ffc954c1ba0 r9:0x7ffc954c1ba0 r10:0x7ffc954c1ba0 r11:0x246 r12:0x55ec46755d60 r13:0x7ffc954c1c80 r14:0x1 r15:0x1 rip:0x70000005 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0x1bc fs_base:0x7f3b01302440 gs_base:0x0
}
{
  real_time:2427115.117039 global_time:200844, event:`SYSCALLBUF_RESET' tid:63895, ticks:86307082
}
{
  real_time:2427115.117554 global_time:200845, event:`SYSCALL: rrcall_notify_syscall_hook_exit' (state:EXITING_SYSCALL) tid:63895, ticks:86307082
 rax:0xf997 rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x0 rdi:0x0 rbp:0xf997 rsp:0x681ffe20 r8:0x7ffc954c1ba0 r9:0x7ffc954c1ba0 r10:0x7ffc954c1ba0 r11:0x246 r12:0x55ec46755d60 r13:0x7ffc954c1c80 r14:0x1 r15:0x1 rip:0x70000005 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0xba fs_base:0x7f3b01302440 gs_base:0x0
}
{
  real_time:2427115.117570 global_time:200846, event:`SIGNAL: SIGUSR1(async)' tid:63895, ticks:86307082
 rax:0xf997 rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x0 rdi:0x0 rbp:0xf997 rsp:0x681ffe20 r8:0x7ffc954c1ba0 r9:0x7ffc954c1ba0 r10:0x7ffc954c1ba0 r11:0x246 r12:0x55ec46755d60 r13:0x7ffc954c1c80 r14:0x1 r15:0x1 rip:0x70000005 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0xba fs_base:0x7f3b01302440 gs_base:0x0
}
{
  real_time:2427115.117633 global_time:200847, event:`SIGNAL_HANDLER: SIGUSR1(async)' tid:63895, ticks:86307082
 rax:0x0 rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x681ff9c0 rsi:0x681ffaf0 rdi:0xa rbp:0xf997 rsp:0x681ff9b8 r8:0x7ffc954c1ba0 r9:0x7ffc954c1ba0 r10:0x7ffc954c1ba0 r11:0x246 r12:0x55ec46755d60 r13:0x7ffc954c1c80 r14:0x1 r15:0x1 rip:0x55ec467561c1 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0xba fs_base:0x7f3b01302440 gs_base:0x0 st0:0x0 st1:0x0 st2:0x0 st3:0x0 st4:0x0 st5:0x0 st6:0x0 st7:0x0ymm0:0x0 ymm1:0x0 ymm2:0x0 ymm3:0x0 ymm4:0x0 ymm5:0x0 ymm6:0x0 ymm7:0x0 ymm8:0x0 ymm9:0x0 ymm10:0x0 ymm11:0x0 ymm12:0x0 ymm13:0x0 ymm14:0x0 ymm15:0x0
}
{
  real_time:2427115.117646 global_time:200848, event:`SIGNAL: SIGALRM(async)' tid:63895, ticks:86307082
 rax:0x0 rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x681ff9c0 rsi:0x681ffaf0 rdi:0xa rbp:0xf997 rsp:0x681ff9b8 r8:0x7ffc954c1ba0 r9:0x7ffc954c1ba0 r10:0x7ffc954c1ba0 r11:0x246 r12:0x55ec46755d60 r13:0x7ffc954c1c80 r14:0x1 r15:0x1 rip:0x55ec467561c1 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0xba fs_base:0x7f3b01302440 gs_base:0x0
}
{
  real_time:2427115.117750 global_time:200849, event:`SIGNAL_DELIVERY: SIGALRM(async)' tid:63895, ticks:86307082
 rax:0x0 rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x681ff9c0 rsi:0x681ffaf0 rdi:0xa rbp:0xf997 rsp:0x681ff9b8 r8:0x7ffc954c1ba0 r9:0x7ffc954c1ba0 r10:0x7ffc954c1ba0 r11:0x246 r12:0x55ec46755d60 r13:0x7ffc954c1c80 r14:0x1 r15:0x1 rip:0x55ec467561c1 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0xba fs_base:0x7f3b01302440 gs_base:0x0
}
```

**Comment by Keno** [comment]
log from the first failure is essentially identical.

**Comment by rocallahan** [comment]
Hmm looks to me like the signal is actually blocked in the tracee? Maybe what's happening is that we get SIGUSR1 and SIGALRM, and then we inject SIGUSR1, which invokes the handler and starts blocking SIGALRM since SIGALRM is in the sigaction mask, so then we try to inject SIGALRM and nothing happens.

**Comment by rocallahan** [comment]
I can probably write a small testcase for this

**Comment by rocallahan** [comment]
I can, after fixing another bug. But now, how to fix...

Normally we'd process SIGUSR1 in such a way that it's delivered and its `sa_mask` is applied before SIGALRM is dequed and everything's good. The problems arise when both signals arrive while we're in syscallbuf processing. We can't deliver SIGUSR1 at that point so there's nothing to stop SIGALRM being dequed by the kernel.

We could try to block other signals from being delivered in the syscallbuf after we've received one, by manipulating the task's signal mask. That might get hairy since we need to undo that (presumably in `SYS_rrcall_notify_syscall_hook_exit`), but what if the syscallbuf calls `sigprocmask` in the meantime...

**Comment by rocallahan** [comment]
Another option is to ignore stashed signals that are currently blocked, essentially blocking them in rr. Problem then is that syscallbuf'ed sigprocmask might unblock the signal at any point, in which case we need to deliver the signal right away, but the task won't be interrupted. I suppose we could add some kind of state flag to tell syscallbuf to trace all sigprocmask syscalls, but that's ugly...

**Comment by rocallahan** [comment]
Yet another option is, if we detect a stashed signal is blocked, unstash it and send it to the process, i.e. move it back to the kernel's signal queue. Then it will go again through the normal signal handling process once it become unblocked. This is probably the way to go.

**Comment by rocallahan** [comment]
One extra wrinkle is that when a signal is sent, it can be sent to a specific task or a thread group and we generally don't know which it was. If sent to a thread group, the kernel might decide to deliver it to the running task T while another thread T2 could also handle the signal. Then rr stashes the signal for thread T, and later discovers when trying to deliver the signal that T is blocking the signal now. Really we should then send it to T2 instead but we don't know, it depends on whether it was send to T or the thread group and AFAIK there's no way to tell :-(

**Comment by rocallahan** [comment]
I don't think that particular problem can be fixed except by updating the signal mask eagerly.

**Comment by rocallahan** [comment]
There is `PTRACE_PEEKSIGINFO` but I'm not sure if that's useful here.

**Comment by Keno** [comment]
Would it make sense to apply the sa_mask effects when the signal arrives, but not actually run the signal handler until we're out of the syscallbuf code?

**Comment by rocallahan** [comment]
Yeah, that's what I'm getting at.

We have a bug related to https://github.com/mozilla/rr/issues/1965#issuecomment-276921422 exists independent of `sa_mask` because of syscallbuf'ed sigprocmask. If a signal arrives in the syscallbuf before sigprocmask, then sigprocmask runs to block the signal, we'll later try to deliver it and fail. We actually have code to prevent this bug with non-syscallbufed sigprocmask in `interrupt_syscall_to_handle_signals`: we fake interruption of the syscall in order to deliver all stashed signals before the sigprocmask takes effect.

**Comment by rocallahan** [comment]
So how much of a perf win was sigprocmask buffering again? :-)

**Comment by Keno** [comment]
A lot :/

**Comment by rocallahan** [comment]
Ignoring syscallbufed sigprocmasks for a moment, I think a fix for this bug would be:
* Make `Task::resume_execution` temporarily set the task's sigprocmask to block all (non-rr) signals, via `PTRACE_SETSIGMASK`, whenever there are stashed signals (unapplying at each task-stop)
* When we enter a traced syscall, if we have a stashed signal, fake a syscall interruption and deliver the signal (similar to what we currently do for `exit` and `sigprocmask` currently).
I think this would actually ensure there's only ever one stashed signal. This should work because the real sigprocmask value only matters when we a) execute a syscall that depends on the sigprocmask value or b) block at a point where a signal should be delivered. The above approach would guarantee that our block-all-signals never happens while we're executing a syscall so neither of those will apply.

**Comment by rocallahan** [comment]
> Make `Task::resume_execution`
Actually, something slightly higher-level since it doesn't know about stashed signals.

**Comment by rocallahan** [comment]
As for syscallbufed sigprocmask, maybe this:
* While there's a stashed signal, (as well as blocking all signals) set a temporary breakpoint on the untraced-syscall entry point that sigprocmask uses. Remove the breakpoint at each stop.
* Treat sigprocmask as a MAY_BLOCK syscall
* When we hit that breakpoint, if it's a sigprocmask syscall, fake an interruption of the sigprocmask syscall and proceed as if we'd received a desched signal: deliver the stashed signal, then allow sigprocmask to proceed.

**Comment by rocallahan** [comment]
I think we should release rr 4.5.0 before fixing this bug. Normally I like to fix all reproducible outstanding bugs before a release, but fixing this is likely to introduce some risk, so I wouldn't feel without some amount of user testing of master between fixing this bug and doing a release. The underlying bug here has been present for a very long time so I don't think it represents a regression since 4.4.0. I want to do an rr 4.5.0 release now because it contains workarounds for the Linux mm regression introduced by the DirtyCOW fix that has been widely backported (plus a lot of other good fixes).

**Comment by Keno** [comment]
Sounds good to me.

**Comment by Keno** [comment]
I should also point out that my patch for that regression was accepted and is making its way back into stable kernels.

**Comment by rocallahan** [comment]
oh good. I still think this is a good time to release rr 4.5.0 now that I've almost completed testing and fixing the bugs that turned up :-)

**Comment by rocallahan** [comment]
This is hairy but I seem to be getting somewhere.

**Comment by rocallahan** [comment]
I'm depending on the assumption that when rr tries to inject a signal at a signal-stop, the kernel won't ever respond by notifying us of another signal, so we don't need to try to block (other) signals during the injection. (Doing that is problematic because it means we can't observe the kernel's changes to the signal mask to due to sa_mask etc so we can't test that our assumptions about those changes are correct.)

**Comment by rocallahan** [comment]
This fake restart approach is hugely painful when applied to syscalls that the kernel doesn't consider interruptible (and hence restartable). :-( I'm going to try a slightly different approach which doesn't involve treating uninterruptible syscalls as interruptible:

* While there's a stashed signal, make `Task::resume_execution` temporarily set the task's sigprocmask to block all (non-rr) signals, via `PTRACE_SETSIGMASK` (unapplying at each task-stop)
* While there's a stashed signal, set temporary breakpoints on the (unprivileged) traced and untraced syscall entry points. Do not allow any traced or sigprocmask syscall to be entered while there are stashed signals. Remove breakpoints at each stop. If an untraced non-sigprocmask syscall triggers a breakpoint, disable the breakpoints and allow it to complete normally (we'll deliver the signal(s) at the `SYS_rrcall_notify_syscall_hook_exit` as now).
* When a traced-syscall breakpoint is hit, allow stashed signals to be delivered at that point. We're not in a syscallbuf transaction so this can be done normally.
* When a sigprocmask untraced-syscall breakpoint is hit, don't try to deliver signals there, just emulate sigprocmask magically returning EAGAIN. In `sys_sigprocmask`, commit that result and retry via a traced-syscall, which is guaranteed to dispatch signals and then complete.

**Comment by rocallahan** [comment]
This approach really is a lot easier. As a bonus it lets us remove the existing hack for fake-interrupting exit/sigprocmask.

Testing this I uncovered another existing bug: if a signal is received in syscallbuf code and then we go into a blocking untraced syscall, the signal is not handled until that syscall completes. Surprised that hasn't burned people already. Maybe it has...

**Comment by rocallahan** [comment]
Aaaaaand another existing bug. The current scheme using `SYS_rrcall_notify_syscall_hook_exit` called in `syscall_hook` when `notify_on_syscall_hook_exit` is set doesn't work properly when a signal is received after the `notify_on_syscall_hook_exit` check but while we're still in syscallbuf code (as defined by `RecordTask::is_in_syscallbuf`, i.e. including dynamic jump stubs). The signal gets stashed but won't be delivered until something else stops the tracee outside the syscallbuf (and we might buffer more syscalls in the meantime). With the changes here, signals are blocked that whole time.

I don't remember why I failed to consider that race condition. Maybe I did and decided it was innocuous.

Not sure how I'm going to fix it yet. One option would be to remove the `notify_on_syscall_hook_exit` mechanism entirely, create a way to get the return address of the current syscallbuf activation, and extend the new breakpoint mechanism to set a breakpoint at that return address.That will let us stop at exactly the first instruction outside the syscallbuf and deliver signals there.

**Comment by rocallahan** [comment]
There's a related version of that issue where you call one of the functions overridden in preload.c (e.g. `sysconf`, `pthread_create`) and a signal arrives and is stashed, then we leave the function without applying the signal.

**Comment by rocallahan** [comment]
I think I might be able to use a custom linker script to put all the code to be treated as non-interruptible by signals into a single region whose bounds can then be exposed to rr.

**Comment by rocallahan** [comment]
Instead of using a custom linker script I can use section attributes to place functions in specific functions, and then rely on the linker not reordering functions too much to mostly get what I need.

Unfortunately I've discovered that we call out from syscallbuf code to libc code --- or preload code that's shared between syscallbuf code and non-syscallbuf overrides --- in some difficult-to-spot places. Namely:
* Usage of `errno` (turns into call to `__errno_location`)
* Usage of `vsnprintf`
* Usage of `strerror`
* Usage of `strlen`
* Usage of `__x86.get_pc_thunk` stubs. These are in our library but shared between syscallbuf and non-syscallbuf code.

**Comment by rocallahan** [comment]
The first four can all be eliminated fairly easily if we just lobotomize the logging, which is fine because it's hardly ever important.

The last one is a bit of a problem. If we treat those thunks as syscallbuf code, then when non-syscallbuf code calls into them, a signal could arrive and we'll block all signals then return to the non-syscallbuf code and wreak havoc. If we treat them as non-syscallbuf code, then syscallbuf code can call into them and a signal can be delivered there, violating the atomicity expectations of non-syscallbuf code.

What's particularly disturbing is that in some functions these thunks are called to load the current PC into `ebx` and it seems to never be used, so I don't know how to stop the thunk from being called :-(.

**Comment by rocallahan** [comment]
This is for the 32-bit library only of course.

**Comment by rocallahan** [comment]
I think I can deal with the `__x86.get_pc_thunk` stubs by declaring them myself in an assembly file that's linked before all other files. The linker will throw away all other copies because they're COMDAT. Then we can add logic to rr so that if the current IP is in a get_pc_thunk, we look at the return address to determine if we're in syscallbuf code or not.

**Comment by rocallahan** [comment]
OK, this should fix it: https://github.com/mozilla/rr/commit/f87cc0acb22b64daf50d8936d0dc4eb6bffbefb3

**Comment by Keno** [comment]
Wow, that's quite a chunk of work! Let me pull this and run the test suite as many times as I can to verify.

**Comment by rocallahan** [comment]
It passes tests locally of course, but it may have latent issues.

**Comment by Keno** [comment]
The new tests are failing for me intermittently. I don't have time to investigate this right this moment, but this is on anubis (which you should have access to), so if you want to investigate, I can chown the trace directory to you. Let me know.
```
ctest -I 561,561 --verbose
UpdateCTestConfiguration  from :/data/keno/rr-build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/data/keno/rr-build/DartConfiguration.tcl
Test project /data/keno/rr-build
Constructing a list of tests
Done constructing a list of tests
Checking test dependency graph...
Checking test dependency graph end
test 561
    Start 561: syscallbuf_signal_blocking

561: Test command: /bin/bash "/data/keno/rr/src/test/basic_test.run" "-b" "/data/keno/rr" "/data/keno/rr-build" "syscallbuf_signal_blocking"
561: Test timeout computed to be: 120
561: /data/keno/rr/src/test/util.sh: line 225:  5418 Aborted                 _RR_TRACE_DIR="$workdir" rr $GLOBAL_OPTIONS replay -a $replayflags > replay.out 2> replay.err
561: Test 'syscallbuf_signal_blocking' FAILED: : error during replay:
561: --------------------------------------------------
561: [FATAL /data/keno/rr/src/ReplaySession.cc:886:check_ticks_consistency() errno: SUCCESS]
561:  (task 5420 (rec:5392) at time 265)
561:  -> Assertion `ticks_now == trace_ticks' failed to hold. ticks mismatch for 'SYSCALL: read'; expected 3581106, got 10051226
561: [FATAL /data/keno/rr/src/log.cc:289:emergency_debug() errno: SUCCESS] (session doesn't look interactive, aborting emergency debugging)
561: --------------------------------------------------
561: replay.out:
561: --------------------------------------------------
561: --------------------------------------------------
561: Test syscallbuf_signal_blocking failed, leaving behind /tmp/rr-test-syscallbuf_signal_blocking-EcSINs1Du
561: To replay the failed test, run
561:   _RR_TRACE_DIR=/tmp/rr-test-syscallbuf_signal_blocking-EcSINs1Du rr replay
1/1 Test #561: syscallbuf_signal_blocking .......***Failed  Error regular expression found in output. Regex=[FAILED]  0.22 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.33 sec

The following tests FAILED:
	561 - syscallbuf_signal_blocking (Failed)
```
```
ctest -I 1605,1605 --verbose
UpdateCTestConfiguration  from :/data/keno/rr-build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/data/keno/rr-build/DartConfiguration.tcl
Test project /data/keno/rr-build
Constructing a list of tests
Done constructing a list of tests
Checking test dependency graph...
Checking test dependency graph end
test 1605
    Start 1605: two_signals_with_mask-32

1605: Test command: /bin/bash "/data/keno/rr/src/test/basic_test.run" "-b" "/data/keno/rr" "/data/keno/rr-build" "two_signals_with_mask_32"
1605: Test timeout computed to be: 120
1605: /data/keno/rr/src/test/util.sh: line 225: 18159 Aborted                 _RR_TRACE_DIR="$workdir" rr $GLOBAL_OPTIONS replay -a $replayflags > replay.out 2> replay.err
1605: Test 'two_signals_with_mask_32' FAILED: : error during replay:
1605: --------------------------------------------------
1605: [FATAL /data/keno/rr/src/ReplaySession.cc:886:check_ticks_consistency() errno: SUCCESS]
1605:  (task 18160 (rec:18129) at time 450)
1605:  -> Assertion `ticks_now == trace_ticks' failed to hold. ticks mismatch for 'SYSCALL: read'; expected 3131947, got 10058518
1605: [FATAL /data/keno/rr/src/log.cc:289:emergency_debug() errno: SUCCESS] (session doesn't look interactive, aborting emergency debugging)
1605: --------------------------------------------------
1605: replay.out:
1605: --------------------------------------------------
1605: --------------------------------------------------
1605: Test two_signals_with_mask_32 failed, leaving behind /tmp/rr-test-two_signals_with_mask_32-XCLmgWGec
1605: To replay the failed test, run
1605:   _RR_TRACE_DIR=/tmp/rr-test-two_signals_with_mask_32-XCLmgWGec rr replay
1/1 Test #1605: two_signals_with_mask-32 .........***Failed  Error regular expression found in output. Regex=[FAILED]  0.25 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.36 sec

The following tests FAILED:
	1605 - two_signals_with_mask-32 (Failed)
Errors while running CTest
```

**Comment by rocallahan** [comment]
> so if you want to investigate, I can chown the trace directory to you. Let me know.

Yes, please do!

**Comment by rocallahan** [comment]
Never mind, I was able to reproduce it myself on anubis.

**Comment by rocallahan** [comment]
I'm going to move the Anubis failure to a new issue.

**Comment by rocallahan** [comment]
Filed #1979 for the `syscallbuf_signal_blocking` failure. I can't reproduce `two_signals_with_mask`, not even on `anubis`, but it looks like the same issue as #1979.
